kivymd/uix/behaviors/elevation.py

```diff
@@ -1145,37 +1145,25 @@ def __init__(self, **kwargs):
         self._update_shadow(self, self.elevation)
         super().__init__(**kwargs)
 
-        self._update_shadow_pos(ins, val)
-
-    def _update_shadow(self, instance, value):
-        self._update_shadow_pos(instance, value)
+    def _update_shadow(self, *args):
         if self._elevation > 0:
-            # dynamic elecation position for the shadow
-            if self.shadow_pos == [0, 0]:
-                self._shadow_pos = [0, -self._elevation * 0.4]
-            # HARD Shadow
-            offset = int(dp(self.hard_shadow_offset))
-            size = [
-                int(self.size[0] + (offset * 2)),
-                int(self.size[1] + (offset * 2)),
-            ]
-            im = BytesIO()
-            # Context
-            img = Image.new("RGBA", tuple(size), color=(0, 0, 0, 0))
-            # Draw context
-            shadow = ImageDraw.Draw(img)
-            self.draw_shadow()(
-                [offset, offset],
-                [
-                    int(size[0] - 1 - offset),
-                    int(size[1] - 1 - offset),
-                ],
-                context=shadow
-                # context=ref(shadow)
-            )
-            img = img.filter(
-                ImageFilter.GaussianBlur(
-                    radius=int(dp(1 + self.hard_shadow_offset / 3))
+            # Set shadow size.
+            ratio = self.width / (self.height if self.height != 0 else 1)
+            if -2 < ratio < 2:
+                self._shadow = MDApp.get_running_app().theme_cls.quad_shadow
+                width = soft_width = self.width * 1.9
+                height = soft_height = self.height * 1.9
+            elif ratio <= -2:
+                self._shadow = MDApp.get_running_app().theme_cls.rec_st_shadow
+                ratio = abs(ratio)
+                if ratio > 5:
+                    ratio = ratio * 22
+                else:
+                    ratio = ratio * 11.5
+                width = soft_width = self.width * 1.9
+                height = self.height + dp(ratio)
+                soft_height = (
+                    self.height + dp(ratio) + dp(self._elevation) * 0.5
```

Comment:
Are we back to old changes?

Reply:
more like mockup shadows for bigger elements, 
the new shadows render the background dynamically, the old draws a mockup texture.
while the new shadows cast a better and 
stylized shadow and elevation. you can see how the shadows differ between the Fake**elevationbehavior and the Rectangular/circularElevarion behavior in the videos above.

Reply:
While material IO ask for a dynamic shadow cast, stating that shadow should change with the widget, fake shadows are a great tool for most static widgets.